### PR TITLE
feat: update db image versions in docker compose

### LINKED ISSUE #9326: Update docker image that generated by typeorm cli: `npx typeorm --init docker`
### The Problem

Docker image that generated by TypeORM CLI is old.

example:
- `npx typeorm --init docker`
- open docker-compose.yml
```yaml
version: '3'
services:

  postgres:
    image: "postgres:9.6.1"
    ports:
      - "5432:5432"
    environment:
      POSTGRES_USER: "test"
      POSTGRES_PASSWORD: "test"
      POSTGRES_DB: "test"
```

I think `image: "postgres:9.6.1"` is old. So I want to update.

### The Solution

Update `getDockerComposeTemplate()` to use `latest`.

https://github.com/typeorm/typeorm/blob/master/src/commands/InitCommand.ts#L521

### Relevant Database Driver(s)

| DB Type           | Relevant      |
|-------------------| ---           |
| `aurora-mysql`    | no            |
| `aurora-postgres` | no            |
| `better-sqlite3`  | no            |
| `cockroachdb`     | no            |
| `cordova`         | no            |
| `expo`            | no            |
| `mongodb`         | no            |
| `mysql`           | no            |
| `nativescript`    | no            |
| `oracle`          | no            |
| `postgres`        | no            |
| `react-native`    | no            |
| `sap`             | no            |
| `spanner`         | no            |
| `sqlite`          | no            |
| `sqlite-abstract` | no            |
| `sqljs`           | no            |
| `sqlserver`       | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✅ Yes, I have the time, and I know how to start.
 - ✖️ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

### PR DESCRIPTION
**Pull request by Jacse**
### Description of change

Fixes #9326

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run format` to apply prettier formatting
- [ ] `npm run test` passes with this change
- [x] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Jacse** [comment]
Note: I haven't been able to functionally test this on my ARM mac.

**Comment by pleerock** [comment]
Please answer my comments from #9366 here.

**Comment by Jacse** [comment]
Hi @pleerock I've fixed the postgres version, but I'm having a hard time testing/verifying the versions on my M1 mac... Time is pretty short on my end but i'll try to see if I can get it running in the weekend

File: docker-compose.yml [hunk 1]
```diff
@@ -44,7 +29,7 @@ services:
     # mdillon/postgis is postgres + PostGIS (only). if you need additional
     # extensions, it's probably time to create a purpose-built image with all
     # necessary extensions. sorry, and thanks for adding support for them!
-    image: "mdillon/postgis:9.6"
+    image: "mdillon/postgis:14.5"
```
Comment (Jacse) [inline docker-compose.yml:32]: I'm not sure about changing the version of this postgres versus the postgres-12 that's also here

**Comment by Jacse** [comment]
@pleerock do you have any functional testing environment that I can boot? Or what's the recommended testing strategy?

**Comment by pleerock** [comment]
you can change local docker-compose file, run all new images and run `npm run test` to test everything.

**Comment by Jacse** [comment]
Ok awesome, thanks

**Comment by Jacse** [comment]
All tests pass ☑️

**Comment by pleerock** [comment]
Thank you for contribution! 🎉

**Pull request merged**